biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -353,413 +376,494 @@ subroutine spawn_patches( currentSite, bc_in)
-            cwd_ag_local, cwd_bg_local, leaf_litter_local, &
-            root_litter_local, bc_in%nlevsoil)
-
-       new_patch%tallest  => null()
-       new_patch%shortest => null()
+       ! create two empty patches, to absorb newly disturbed primary and secondary forest area
+       ! first create patch to receive primary forest area
+       if ( site_areadis_primary .gt. nearzero ) then
+          allocate(new_patch_primary)
+          call create_patch(currentSite, new_patch_primary, age, site_areadis_primary, &
+               cwd_ag_local, cwd_bg_local, leaf_litter_local, &
+               root_litter_local, bc_in%nlevsoil, primaryforest)
+          new_patch_primary%tallest  => null()
+          new_patch_primary%shortest => null()
+       endif
 
+       ! next create patch to receive secondary forest area
+       if ( site_areadis_secondary .gt. nearzero) then
+          allocate(new_patch_secondary)
+          call create_patch(currentSite, new_patch_secondary, age, site_areadis_secondary, &
+               cwd_ag_local, cwd_bg_local, leaf_litter_local, &
+               root_litter_local, bc_in%nlevsoil, secondaryforest)
+          new_patch_secondary%tallest  => null()
+          new_patch_secondary%shortest => null()
+       endif
+    
        currentPatch => currentSite%oldest_patch
        ! loop round all the patches that contribute surviving indivduals and litter pools to the new patch.     
        do while(associated(currentPatch))   
 
+
           ! This is the amount of patch area that is disturbed, and donated by the donor
           patch_site_areadis = currentPatch%area * currentPatch%disturbance_rate
-
-          if (patch_site_areadis > nearzero) then
-
-          call average_patch_properties(currentPatch, new_patch, patch_site_areadis)
           
-          if (currentPatch%disturbance_rates(dtype_ilog) > currentPatch%disturbance_rates(dtype_ifall) .and. &
-                currentPatch%disturbance_rates(dtype_ilog) > currentPatch%disturbance_rates(dtype_ifire) ) then 
-             
-             call logging_litter_fluxes(currentSite, currentPatch, new_patch, patch_site_areadis)
-             
-          elseif (currentPatch%disturbance_rates(dtype_ifire) > currentPatch%disturbance_rates(dtype_ifall) .and. &
-                currentPatch%disturbance_rates(dtype_ifire) > currentPatch%disturbance_rates(dtype_ilog) ) then
-             
-             call fire_litter_fluxes(currentSite, currentPatch, new_patch, patch_site_areadis)  
-             
-          else
-             
-             call mortality_litter_fluxes(currentSite, currentPatch, new_patch, patch_site_areadis)
-             
-          endif
-
-          !INSERT SURVIVORS FROM DISTURBANCE INTO NEW PATCH 
-          currentCohort => currentPatch%shortest
-          do while(associated(currentCohort))       
+          if (patch_site_areadis > nearzero ) then
 
              allocate(nc)
```

Comment:
This whole allocation block needs to be removed.

Reply:
yep, good catch.